### PR TITLE
Core/Misc: Update enums from db2 to 9.0.2.37176

### PR DESCRIPTION
**Pull request by osleyder85**
**Changes proposed:**

-  Update CreatureFamily.db2
    In the case of `CREATURE_FAMILY_HORSE_CUSTOM`, check the Database and there is no creature with that family.
-  Update CreatureType.db2 
-  Update Holidays.db2
-  Update LockType.db2
-  Update QuestInfo.db2

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [X] master

**Tests performed:**

(Does it build, tested in-game, etc.)

- [X] It builds
- [X] Tests in game

### CODE REVIEW

**Comment by illfated** [comment]
Looks pretty clean and straightforward to me, but I will leave it to the TC members who use the master branch to validate the content.

**Comment by Shauren** [comment]
These "duplicated" ids are entries for different regions that have resets on different days

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -4258,60 +4276,344 @@ enum CreatureEliteType
...
+    HOLIDAY_NOBLEGARDEN                            = 181,
+    HOLIDAY_CHILDRENS_WEEK                         = 201,
+    HOLIDAY_CALL_TO_ARMS_ALTERAC_VALLEY            = 283,
+    HOLIDAY_CALL_TO_ARMS_WARSONG_GULCH             = 284,
+    HOLIDAY_CALL_TO_ARMS_ARATHI_BASIN              = 285,
+    HOLIDAY_HARVEST_FESTIVAL                       = 321,
+    HOLIDAY_HALLOWS_END                            = 324,
+    HOLIDAY_LUNAR_FESTIVAL                         = 327,
+    // HOLIDAY_LOVE_IS_IN_THE_AIR                     = 335, duplicated
+    HOLIDAY_MINDSUMMER_FIRE_FESTIVAL               = 341,
```
Comment (mdX7) [inline src/server/game/Miscellaneous/SharedDefines.h:4295]: typo? - Midsummer Fire Festival should be correct

Comment (osleyder85) [reply src/server/game/Miscellaneous/SharedDefines.h:4295]: https://wow.tools/dbc/?dbc=holidays&build=9.0.2.37176#page=1&colFilter[0]=341

sdd

Comment (MaxtorCoder) [reply src/server/game/Miscellaneous/SharedDefines.h:4295]: You have `MINDSUMMER`, not `MIDSUMMER` hahah

Comment (osleyder85) [reply src/server/game/Miscellaneous/SharedDefines.h:4295]: hahahaha, I did not see it.

**Comment by osleyder85** [comment]
> 
> 
> These "duplicated" ids are entries for different regions that have resets on different days

@Shauren you think it would be a good idea to put HOLIDAYS_NAME_#REGION, to be able to identify them all.
Example:
```
HOLIDAY_SHADOWLANDS_DUNGEON_EVENT_961    = 1217,
HOLIDAY_SHADOWLANDS_DUNGEON_EVENT_4      = 1218,
HOLIDAY_SHADOWLANDS_DUNGEON_EVENT_24     = 1219,
HOLIDAY_SHADOWLANDS_DUNGEON_EVENT_2      = 1220,
```
If it's okay with you, I can do it like this.

**Comment by Shauren** [comment]
Regions:
1 - US
2 - KR
4 - EU
8 - TW
16 - CN

Yes, its flags

As for descriptions - just read them, its most likely different LFR wings openings

**Comment by Shauren** [comment]
Side note: I dont think a complete holiday enum is neccessary

File: src/server/game/Miscellaneous/SharedDefines.h (hunk 1 above)
Comment (illfated) [reply src/server/game/Miscellaneous/SharedDefines.h]: Funny idea. Thank you. I will keep it on my mind until next summer... 🌞 😄

**Comment by osleyder85** [comment]
I do not know the regions 24,32,768,961 and 1984.

**Comment by Shauren** [comment]
24 = CN_TW
32, 768, 1984 = unknown, region with that mask doesnt exist in Cfg_Regions
961 = "default", bunch of unknown regions + US

**Pull request merged**